application_test.go

```diff
@@ -277,23 +277,17 @@ func TestListApplication(t *testing.T) {
 	assert.Equal(t, len(application.Tasks), 2)
 }
 
-func TestHasApplication(t *testing.T) {
+func TestApplicationDoesntExists(t *testing.T) {
```

Comment:
I think the name is (partially) misleading because the second test in this function does not relate to a non-existing application (but rather a generic error case).

You could rename it to something like  `TestApplicationFailing` or consider merging the test with `TestListApplication`. The latter would be somewhat more consistent with the other tests where a single function covers both erroneous and non-erroneous outcomes.
(Side note: I think `TestListApplication` should be called `TestApplication` because `Application` is the interface method being tested. You may want to consider this if you decide to merge.)